### DISCUSSION TITLE
Number of widgets per row

### DISCUSSION THREAD

**Discussion by oliverbj** [filamentphp/filament#14036] [General]
Hi.

I noticed this PR #13415 which was closed. Are there any way to allow more than three widgets per row?

**Comment by oliverbj** [comment]
Anyone?

**Comment by e2d2-dev** [comment]
Hi,

just use inside a widget (StatWidget)

    public function getColumns(): int
    {
        return 4; // for 4 in a row
    }

There is something similar for the Dashboard.


ciao ed

**Comment by oliverbj** [comment]
Sorry, that doesn't work. Perhaps I am doing something wrong. I have this:

```php
class TopStatsWidget extends BaseWidget
{
    protected static ?int $sort = 0;
    
    protected int|string|array $columnSpan = 12;

    protected function getStats(): array
    {
        $data = $this->fetchData();
        $currency = $this->getCurrency();

        return [
            $this->getAverageJobProfitStat($data, $currency),
            $this->getJobsBelowProfitStat($data, $currency),
            $this->getJobsBelowZeroStat($data, $currency),
            $this->getAverageProductivityStat($data),
            $this->getAverageInvoiceSpeedStat()
        ];
    }

    public function getColumns(): int
    {
        return 5;
    }
```

As I understand, I have to specify the columnspan (i.e., how many columns that widget takes). I set that to `12`. I then set the `getColumns` to 5 (since I have 5 widgets/stats).

In my `Dashboard.php` I have this:

```
class Dashboard extends BaseDashboard
{
    use HasFiltersForm;

    public function getColumns(): int | string | array
    {
        return 12;
    }
}
```

**Comment by oliverbj** [comment]
Anyone?

**Comment by leandrocfe** [comment]
I think status widget is limited to 4 items
https://discord.com/channels/883083792112300104/1217795889741955072/1217795889741955072

**Comment by oliverbj** [comment]
Thank you! I spend a lot of time trying to debug this one.
